app/src/main/kotlin/com/ortin/gesturetranslator/app/presentation/MainActivity.kt

```diff
@@ -0,0 +1,143 @@
+package com.ortin.gesturetranslator.app.presentation
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.util.Log
+import android.view.MenuItem
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import androidx.navigation.fragment.NavHostFragment
+import com.ortin.gesturetranslator.R
+import com.ortin.gesturetranslator.databinding.ActivityMainBinding
+import com.ortin.gesturetranslator.domain.usecases.SaveLoadSettingsUseCase
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class MainActivity : AppCompatActivity() {
+    @Inject
+    lateinit var saveLoadSettingsUseCase: SaveLoadSettingsUseCase
+
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var navController: NavController
+//    val callback = this.onBackPressedDispatcher.addCallback(this.getLifecycle())
+
+    @SuppressLint("SourceLockedOrientationActivity")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+
+        if (saveLoadSettingsUseCase.getTheme()) {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+        } else {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+        }
+        setContentView(binding.root)
+
+        init()
+        initListeners()
+    }
+
+    private fun init() {
+        navController = (supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment).navController
```

Comment:
```suggestion
        navController = findNavController(binding.hostFragmentView)
```

Замени hostFragmentView на свой

Reply:
Исправил